src/components/ui/HeaderUserMenu.js

```diff
@@ -1,48 +1,73 @@
-import RestrictedComponent from 'components/logic/RestrictedComponent'
-import React, { useState } from 'react'
+import { Avatar } from "@chakra-ui/avatar";
+import { ChevronDownIcon } from "@chakra-ui/icons";
+import { Badge, Divider, HStack, Text } from "@chakra-ui/layout";
+import { Menu, MenuButton, MenuItem, MenuList } from "@chakra-ui/menu";
+import RestrictedComponent from "components/logic/RestrictedComponent";
+import { AuthConsumer } from "contexts/AuthContext";
+import React from "react";
 import { Link } from 'react-router-dom'
-import { AuthConsumer } from '../../contexts/AuthContext'
-import UserAvatar from './../badges/UserAvatar'
-import './HeaderUserMenu.scss'
+import MD5 from "services/md5";
 
 export default function HeaderUserMenu({ email }) {
-
-    const [showWindow, setShowWindow] = useState(false)
-    const handleShowWindow = () => {
-        setShowWindow(!showWindow)
-        setTimeout(() => {
-            setShowWindow(false)
-        }, 10000)
-    }
-
+    const user = JSON.parse(localStorage.getItem("user"));
     return (
-        <div style={{ position: 'relative' }}>
-            <UserAvatar onClick={handleShowWindow} email={email} />
-            {showWindow && <UserMenu />}
-        </div>
-    )
-}
-
-
-const UserMenu = () => {
-    const user = JSON.parse(localStorage.getItem('user'));
+        <Menu closeOnBlur closeOnSelect>
+            <MenuButton rightIcon={<ChevronDownIcon />}>
+                <Avatar
+                    name={user?.name}
+                    size="sm"
+                    backgroundColor="gray.700"
+                    src={`https://www.gravatar.com/avatar/${MD5(
+                        email
+                    )}?s=200&d=robohash`}
+                />
```

Comment:
The component `UserAvatar` is being used in a few different places. To avoid repeating this settings again and again modify the internal `UserAvatar`implementation and in one go we have it done.